### PR TITLE
feat: create ChipButton element

### PR DESCRIPTION
**Pull request by berteotti**
This PR adds the new ChipButton and RadioButton UI elements.

Both components are present in UI page.

image

### CODE REVIEW

**Comment by Diogomartf** [comment]
the difference between active and disabled is very subtle. Don't know if would be better to add more constrast? What do you think?

image

File: packages/app/ui/buttons/RadioButton.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+export const radioButtonStyles = cva(
+  [
+    "flex items-center justify-center",
+    "cursor-pointer select-none font-medium",
+    "active:ring-4",
+    "disabled:bg-surface-75 disabled:text-em-disabled disabled:cursor-not-allowed disabled:ring-0",
+    "focus:outline-none focus:ring-4",
+    "rounded-full px-3 py-2 space-x-1.5 text-xs",
```
Comment (Diogomartf) [inline packages/app/ui/buttons/RadioButton.tsx:16]: This is just a suggestion. 

I would prefer to move this string up to the second element, near the `flex items-center`.
I like to have the padding above.

Also `text-xs` could be near `font-medium` as they are relatable as well.

Not important, just sharing a preference.

File: packages/app/ui/buttons/RadioButton.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        value={value}
+        id={id}
+        onChange={onChange}
+        checked={checked}
+        name={name}
+        className="w-0 h-0"
+      />
+      <label
+        htmlFor={id}
+        className={cx(
```
Comment (Diogomartf) [inline packages/app/ui/buttons/RadioButton.tsx:59]: what styles are you joining here?

I think there is only `radioButtonStyles` so you can remove `cx`.

File: packages/app/ui/buttons/ChipButton.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  onClick,
+  active,
+  id,
+}: ChipButtonProps) {
+  return (
+    <button
+      id={id}
+      onClick={onClick}
+      disabled={disabled}
+      className={twMerge(
```
Comment (Diogomartf) [inline packages/app/ui/buttons/ChipButton.tsx:33]: why use `twMerge` with `cx` ?

**Comment by Diogomartf** [comment]
75db7299d99ebbd2d79f22f117c766d0

wondering if it wouldn't be better if the `active ring` when clicking an inactive radio btn was green (primary color) instead of gray?

File: packages/app/ui/buttons/ChipButton.tsx (hunk 3 above)
Comment (berteotti) [reply packages/app/ui/buttons/ChipButton.tsx:33]: There were some conflicts happening, so had to use `twMerge` here. `cx` wasn't handling conflicts, also they recommend it on their docs https://cva.style/docs/installation#handling-style-conflicts

File: packages/app/ui/buttons/RadioButton.tsx (hunk 1 above)
Comment (berteotti) [reply packages/app/ui/buttons/RadioButton.tsx:16]: Maybe we should have some guidelines to better define how to organize classes.

**Comment by berteotti** [comment]
> the difference between active and disabled is very subtle. Don't know if would be better to add more constrast? What do you think?
> 
> image

In the design case, I think it's desired to be subtle, almost as a "this isn't an option". It's clearly not disabled as the text and cursor show it's enabled. But we can improve. What do you think it should be?

image

File: packages/app/ui/buttons/RadioButton.tsx (hunk 1 above)
Comment (Diogomartf) [reply packages/app/ui/buttons/RadioButton.tsx:16]: Maybe we should. We can also try it loose for a couple more weeks and then decide the guidelines.

File: packages/app/app/ui/page.tsx [hunk 4]
```diff
@@ -28,13 +38,13 @@ export default function Page() {
...
-            active
+            Active
           </Button>
           <Button size="sm" disabled={true} onClick={() => console.log("hey")}>
-            button disabled
+            Button disabled
           </Button>
         </UISubSection>
-        <UISubSection title="secondary">
+        <UISubSection title="Secondary">
```
Comment (Diogomartf) [inline packages/app/app/ui/page.tsx:47]: maybe `UISection` and `UISubSection` should have the style capitalize instead of us making it.

**Comment by berteotti** [comment]
> 75db7299d99ebbd2d79f22f117c766d0
> 
> wondering if it wouldn't be better if the `active ring` when clicking an inactive radio btn was green (primary color) instead of gray?

I think the active should reflect the current state before clicking. Mixing colors doesn't sound good to me.

File: packages/app/app/ui/page.tsx (hunk 4 above)
Comment (berteotti) [reply packages/app/app/ui/page.tsx:47]: It's just a detail. I like it capitalize but we can also have it normal. I think the components should control it, so we can have the freedom to do it as we want :)

**Comment by Diogomartf** [comment]
> > the difference between active and disabled is very subtle. Don't know if would be better to add more constrast? What do you think?
> > image
> 
> In the design case, I think it's desired to be subtle, almost as a "this isn't an option". It's clearly not disabled as the text and cursor show it's enabled. But we can improve. What do you think it should be?
> 
> image

Don't have a good suggestion right now, let it be, and we see if it works in the wild and if not we improve it in the future.

**Review by Diogomartf — APPROVED** [review]
Good work Leo 👍

**Pull request merged**